Add FAQ: lingering agent pods. (#139)
metalbear-co/mirrord.dev: content/docs/overview/faq.md

No, mirrord doesn't install anything on the cluster, nor does it have any persistent state. It does spawn a short-living pod/container to run the proxy, which is automatically removed when mirrord exits. mirrord works using the Kubernetes API, and so the only prerequisite to start using mirrord is to have kubectl configured for your cluster.

If you have any restrictions for pulling external images inside your cluster, you have to allow pulling of ghcr.io/metalbear-co/mirrord image.

> ## Old mirrord agent pods are not getting deleted after the mirrord run is completed, what can I do?

Reviewer: Please move it to the bottom of the list

Author: Done.

No, mirrord doesn't install anything on the cluster, nor does it have any persistent state. It does spawn a short-living pod/container to run the proxy, which is automatically removed when mirrord exits. mirrord works using the Kubernetes API, and so the only prerequisite to start using mirrord is to have kubectl configured for your cluster.

If you have any restrictions for pulling external images inside your cluster, you have to allow pulling of ghcr.io/metalbear-co/mirrord image.

## Old mirrord agent pods are not getting deleted after the mirrord run is completed, what can I do?

If an agent pod's status is `Running`, it means mirrord is probably still running locally as well. Once you
terminate the local process, the agent pod's status should change to `Completed`.

> On clusters with Kubernetes version v1.23 or newer, agent pods are supposed to be

Reviewer: ```suggestion
On clusters with Kubernetes version v1.23 or newer, agent pods are
```

> [cleaned up automatically](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/) after a

Reviewer: Rephrase to: automatically cleaned up immediately (or after a [configurable TTL](link)).

Author: Done in https://github.com/metalbear-co/mirrord.dev/pull/139/commits/b29b638ecd69c479313efc6a706f760296f4191e

> [configurable](/docs/overview/configuration/#agent-ttl) TTL from completion.

Reviewer 2: I would add that by default that immediately.

Author: Like this? https://github.com/metalbear-co/mirrord.dev/pull/139/commits/5137ed8d6b67a5a6354429b76160c60c79d83c7f

> If on your cluster mirrord agent pods are lingering for

Reviewer: ```suggestion
If your cluster is v1.23 or higher and mirrord agent pods are not being cleaned up automatically
```

> If your cluster is v1.23 or higher and mirrord agent pods are not being cleaned up automatically
> longer than the configured TTL after completion,

Author: ```suggestion
If your cluster is v1.23 or higher and mirrord agent pods are not being cleaned up automatically,
```

> longer than the configured TTL after completion,

Author: removed (https://github.com/metalbear-co/mirrord.dev/pull/139/commits/5f6090d699b8aa666f2b9ad13f94027edb877bef)

[please open an issue on GitHub](

> https://github.com/metalbear-co/mirrord/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml&title=Agent%20pods%20lingering%20after%20completion

Author: Is the issue title ok? "Agent pods lingering after completion"

Reviewer 2: Nice
